pulp_rpm/app/modelresource.py

```diff
@@ -74,6 +74,9 @@ def set_up_queryset(self):
 
         return content
 
+    def dehydrate__pulp_domain(self, content):
```

Comment:
Once this is in QueryModelResource, we can lose it from here.

Reply:
Well there's a distinction between `dehydrate_pulp_domain()` and `dehydrate__pulp_domain()`.  The first one makes sense to put in `QueryModelResource` because it's used throughout Pulpcore.  The second one only makes sense to put there knowing that it's a convention used in plugins.

I'm not against doing that and I think it could make sense to put it there just so it can't be forgotten (and it does no harm).  Just calling out that it's not the same.

I am kinda curious how it even works though, it's not using natural key but the actual PK, and I can't find where we export domains?  I can find nothing related in the export tar.

Reply:
ugh, I completely missed the extra `_`. Can you add a comment here pointing that out, it's pretty subtle and (obviously) easy to miss. Only the detail-entities have _pulp_domain, so we can't put it into BaseContentResource, I don't think. I guess "go with this" for now.

We do *not* export domains --  if you try to export on a domain-enabled system it Complains Bitterly at you and refuses. "At some point" we need to make it possible to do so on a domain-enabled system, but there hasn't been a pressing need for it so far. (my take is that one would export per-domain, and import-into a specified domain, so we do *not* move domains themselves across the airgap. fwiw.)

Reply:
>Only the detail-entities have _pulp_domain, so we can't put it into BaseContentResource, I don't think

I don't think it's a blocker, it's not like it does any harm.  I'm pretty sure it would just be a no-op.

Also do we even export "Content" as a non-detail entity?  I don't think we do

Reply:
+1 to docs suggestion, but that will go into Pulpcore